### PR TITLE
add load more and handle search event

### PR DESCRIPTION
**Pull request by anhmv-1205**
## Add load more and handle search event

https://edu-redmine.sun-asterisk.vn/issues/13037

### CODE REVIEW

**Comment by anhmv-1205** [comment]
ready

File: app/src/main/java/com/sun_asterisk/moviedb_44/screen/search/SearchActivity.kt [hunk 1]
```diff
@@ -38,5 +48,22 @@ class SearchActivity : AppCompatActivity() {
...
+                viewModel.loadMore()
+            }
+        })
+    }
+
+    override fun onItemClickListener(data: Movie) {
+    }
+
+    override fun onClick(v: View?) {
+        onBackPressed()
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/sun_asterisk/moviedb_44/screen/search/SearchActivity.kt]: chưa check id mà sao click e

File: app/src/main/java/com/sun_asterisk/moviedb_44/utils/OnItemRecyclerViewClick.kt [hunk 2]
```diff
@@ -0,0 +1,5 @@
+package com.sun_asterisk.moviedb_44.utils
+
+interface OnItemRecyclerViewClick<T> {
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/sun_asterisk/moviedb_44/utils/OnItemRecyclerViewClick.kt]:
```suggestion
interface OnItemRecyclerViewClickListener<T> {
```

File: app/src/main/java/com/sun_asterisk/moviedb_44/utils/OnItemRecyclerViewClick.kt [hunk 3]
```diff
@@ -0,0 +1,5 @@
...
+    fun onItemClickListener(data: T)
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/sun_asterisk/moviedb_44/utils/OnItemRecyclerViewClick.kt]:
```suggestion
    fun onItemClick(data: T)
```

**Comment by anhmv-1205** [comment]
ready

**Review by tuanlvt-1117 — APPROVED** [review]

**Pull request merged**